packages/source-iottwinmaker/src/knowledgeGraph-module/KGDataModule.ts

```diff
@@ -0,0 +1,48 @@
+import {
+  ExecuteQueryCommand,
+  ExecuteQueryCommandOutput,
+  ExecuteQueryCommandInput,
+  IoTTwinMakerClient,
+  ColumnDescription,
+  Row,
+} from '@aws-sdk/client-iottwinmaker';
+import { KGDataModuleInput } from './types';
+import { TwinMakerKGQueryDataModule } from '../types';
+
+export class KGDataModule implements TwinMakerKGQueryDataModule {
+  private twinMakerClient: IoTTwinMakerClient;
+  private workspaceId: string;
+  private queryStatement: string;
+  private maxResultsCount: number;
+
+  constructor(input: KGDataModuleInput) {
+    this.twinMakerClient = input.twinMakerClient;
+    this.workspaceId = input.workspaceId;
+    this.queryStatement = input.queryStatement;
+    if (input.maxResultsCount) this.maxResultsCount = input.maxResultsCount;
+  }
+
+  executeQuery = async (): Promise<ExecuteQueryCommandOutput> => {
+    const rows: Row[] = [];
+    let columnDescriptions: ColumnDescription[] = [];
+    let $metadata = {};
+    const commandInput: ExecuteQueryCommandInput = {
+      workspaceId: this.workspaceId,
+      queryStatement: this.queryStatement,
+    };
+    do {
+      const response: ExecuteQueryCommandOutput = await this.twinMakerClient.send(
+        new ExecuteQueryCommand(commandInput)
+      );
+      if (response.rows) {
+        rows.push(...response.rows);
+      }
+      if (response.columnDescriptions && columnDescriptions.length === 0) {
+        columnDescriptions = response.columnDescriptions;
+      }
+      if (response.$metadata && Object.keys($metadata).length === 0) $metadata = response.$metadata;
+      commandInput.nextToken = response.nextToken;
+    } while (this.maxResultsCount ? commandInput.nextToken && this.maxResultsCount : commandInput.nextToken);
```

Comment:
It doesn't look like we're submitting maxResults as an API parameter or checking it's value to see if it's exceed manually?

Reply:
The maxResults here is to allow user to control large results and the intention is not related to the maxResult API parameter as the results are paginated and this one refer to the max results to be returned per page. The ExecuteQuery Request maxResult param has default 25 and max 250 which the number of results per page. That means if a use sets maxResultsCount=10 will still get 25 results as this is the minimum default according to pagination.

Reply:
https://docs.aws.amazon.com/AWSJavaScriptSDK/v3/latest/clients/client-iottwinmaker/interfaces/executequerycommandinput.html#maxresults

Reply:
so the you will always get multiples of 25 by default if you set maxResultsCount=26 you will still get 50 results. and Thats fine and explained well in the documentation. just to avoid the case when you have 10,000 results this gives you the control to round up to the closest multiple of 25.